File: app/src/main/java/org/exallium/tradetracker/app/model/rest/MtgApiRestService.java
package org.exallium.tradetracker.app.model.rest;

import retrofit.http.GET;
import retrofit.http.Query;
import rx.Observable;

import java.util.Map;

public interface MtgApiRestService {

    @GET("/cards")
    Observable<Map<String, Object>> getCardsForSet(@Query("set") String setCode);

    @GET("/sets")
    Observable<Map<String, Object>> getSets();
}

File: app/src/main/kotlin/org/exallium/tradetracker/app/model/rest/MtgApiRestService.kt
package org.exallium.tradetracker.app.model.rest

import retrofit.http.GET
import retrofit.http.Query
import rx.Observable

public trait MtgApiRestService {

    [GET("/cards")]
    public fun getCardsForSet([Query("set")] setCode: String, [Query("page")] page: Int): Observable<Map<String, Any>>

    [GET("/sets")]
    public fun getSets(): Observable<Map<String, Any>>
}